configs/example/sst/riscv_fs.py

```diff
@@ -104,7 +104,7 @@ def createHiFivePlatform(system):
 
     system.platform.pci_host.pio = system.membus.mem_side_ports
 
-    system.platform.rtc = RiscvRTC(frequency=Frequency("100MHz"))
+    system.platform.rtc = RiscvRTC(frequency=Frequency("10MHz"))
```

Comment:
Why are we doing this?

Reply:
The clocks between the simulators are correctly synchronized. However, when the kernel boots and clocksource changes to the RiscvRTC, the frequency no longer is synchronized, which becomes 10x slower.

Reply:
I don't understand this. In this file the clock speed is set via an argument. How can we guarantee these are in-sync?

Reply:
I looked at the kernel boot time and the simulator's simulation time. There was a difference of 10x. However, I'd still say that we need better tests to verify this. The kernel switches to the RiscvRTC here:

[    0.007518] clocksource: Switched to clocksource riscv_clocksource

Booting the same kernel in a gem5 only setup and gem5 + SST shows the same 10x difference.

Reply:
What do you mean "the frequency becomes 10x slower" What frequency? 

You need to justify why this needs to be different for SST compared to the rest of gem5.

@kunpai had to change this 100MHz to get boot to work. See https://gem5-review.googlesource.com/c/public/gem5/+/64013

I wonder if Kunal remembers why he chose 100MHz

Reply:
1. The clock read by the kernel is 10x slower if the frequency is set to 100MHz. This time is read from the `printk` statements during the kernel boot.
2. The simulation ends ~300 ms (as per stats.txt and also SST), but the `printk` statements in the kernel print the kernel time ~3 sec when the kernel is simulated until an m5 exit is encountered.
3. I tried using 1MHz, which, in my case works (exists with m5 exit from early kernel space) but the kernel clock becomes 10x faster. The default frequency is 1MHz in the RiscvRTC simobject (`src/dev/riscv/HiFive.py`)
```
    # The default 1MHz setting is taken from SiFive's U54MC
    # core complex. Set to other frequencies if necessary.
```
5. The clocks between gem5 and SST are in sync. I tested this by looking at every memory packet that gem5 is sending to SST, and, whether SST is receiving the same packet at the exact same time.

Reply:
> What do you mean "the frequency becomes 10x slower" What frequency?
> 
> You need to justify why this needs to be different for SST compared to the rest of gem5.
> 
> @kunpai had to change this 100MHz to get boot to work. See https://gem5-review.googlesource.com/c/public/gem5/+/64013
> 
> I wonder if Kunal remembers why he chose 100MHz

I believe I chose 100 MHz because:
1. Our original chosen value of 1 MHz did not complete an FS run even after a week of running.
2. The RISCV Board provided in `src/python/gem5/components/boards/riscv_board.py`, initializes the RTC to be 100 MHz as well [https://github.com/gem5/gem5/blob/48a40cf2f5182a82de360b7efa497d82e06b1631/src/python/gem5/components/boards/riscv_board.py#L115C65-L115C65](https://github.com/gem5/gem5/blob/48a40cf2f5182a82de360b7efa497d82e06b1631/src/python/gem5/components/boards/riscv_board.py#L115C65-L115C65)

Comparing IPC values for 100 MHz vs 1 MHz for SE mode did not give much of a difference if I remember correctly.

Reply:
Why is SST different than without SST? 

Is it wrong throughout all of gem5?

Reply:
I don't know why this is different in SST. Also, this only happens in RISCV.  I have not extensively tested ARM version of SST, but from what I remember, this issue is absent in the `arm_example.py`.

I'll verify the time reported via `stats.txt` and the `printk` statements for both gem5 and gem5 + SST setup. I'll use a similar system in both cases. I'll get back with the results.

Reply:
I did an experiment with a minimal system with 1 Timing CPU clocked at 1 GHz and 1 GiB DRAM without caches. I executed `riscv-boot-exit-nodisk` which executes `m5 exit` as the kernel init script. Here are the results: https://docs.google.com/spreadsheets/d/1wT6_dCyDIwVWuKAXJb617DeE8E5rDKZ3wcgOjh1S_l8/edit?usp=sharing

Reply:
Here's the results for those of you who don't want to open a google doc

<html>
<body>
<!--StartFragment--><google-sheets-html-origin><style type="text/css"><!--td {border: 1px solid #cccccc;}br {mso-data-placement:same-cell;}--></style>

Sl. | Experiment | Gem5 | gem5 + SST
-- | -- | -- | --
stats.txt | printk | SST clock | stats.txt | printk
1 | 1MHz | 29.52111 | 2.709169 | 36.4687 | 36.468664 | 3.283364
2 | 10MHz | 30.541303 | 28.106481 | 38.1729 | 38.172885 | 34.528904
3 | 100MHz | 48.81478 | 462.931175 | 75.5352 | 75.535206 | 716.822168

<!--EndFragment-->
</body>
</html>

So, we should change all of these 100MHz to 10MHz, then :)

Can you create a new PR with that change? You don't have to update this PR.

Reply:
Okay, before opening that PR, I just want to test 100MHz with the RISCV board just to be double sure. I'll update you soon with the results.